annotation/tests/test_distribution.py

```diff
@@ -1277,3 +1283,191 @@ def test_get_page_number_combinations(
+    },
+    {
+        "deadline": None,
+        "file_id": 5,
+        "is_validation": False,
+        "job_id": 1,
+        "pages": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        "user_id": "c080408e-0077-4c20-b520-bd4b1541ca60",
+        "status": TASKS_STATUS,
+    },
+    {
+        "deadline": None,
+        "file_id": 3,
+        "is_validation": False,
+        "job_id": 1,
+        "pages": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
+        "user_id": "c080408e-0077-4c20-b520-bd4b1541ca58",
+        "status": TASKS_STATUS,
+    },
+]
+
+EXPECTED_OUTPUT = [
+    [
+        {
+            "user_id": "ba3e0ccf-8661-4ed3-892b-7c291160f631",
+            "default_load": 96,
+            "overall_load": 0,
+        },
+        {
+            "user_id": "c080408e-0077-4c20-b520-bd4b1541ca59",
+            "default_load": 0,
+            "overall_load": 50,
+        },
+    ],
+    [
+        {
+            "user_id": "6e2ce9ac-2fe8-4cc7-bdcd-bac90faa9247",
+            "default_load": 95,
+            "overall_load": 0,
+        },
+        {
+            "user_id": "8c8a333e-d19a-492a-9e78-5df4bec0ec8b",
+            "default_load": 92,
+            "overall_load": 0,
+        },
+        {
+            "user_id": "ba38c547-f7f2-4ece-b56f-9926e34b159a",
+            "default_load": 90,
+            "overall_load": 0,
+        },
+        {
+            "user_id": "c080408e-0077-4c20-b520-bd4b1541ca56",
+            "default_load": 100,
+            "overall_load": 50,
+        },
+        {
+            "user_id": "c080408e-0077-4c20-b520-bd4b1541ca57",
+            "default_load": 100,
+            "overall_load": 100,
+        },
+    ],
+]
+
+
+@pytest.mark.parametrize(
+    "split_multipage_doc_setup,"
+    "validation_type,"
+    "annotators,"
+    "validators,"
+    "annotation_tasks,"
+    "expected_output",
+    [
+        (
+            False,
+            ValidationSchema.cross,
+            USERS[0:2] + USERS[7:],
+            USERS[2:7],
+            ANNOTATION_TASKS_CHOOSE_VALIDATORS,
+            EXPECTED_OUTPUT[0],
+        ),
+        (
+            True,
+            ValidationSchema.hierarchical,
+            USERS[0:2] + USERS[7:],
+            USERS[2:7],
+            ANNOTATION_TASKS_CHOOSE_VALIDATORS,
+            EXPECTED_OUTPUT[1],
+        ),
+        (
+            True,
+            ValidationSchema.validation_only,
+            USERS[0:2] + USERS[7:],
+            USERS[2:7],
+            ANNOTATION_TASKS_CHOOSE_VALIDATORS,
+            EXPECTED_OUTPUT[1],
+        ),
+    ],
+)
+@pytest.mark.unittest
+def test_choose_validators_users(
+    split_multipage_doc_setup,
+    validation_type,
+    annotators,
+    validators,
+    annotation_tasks,
+    expected_output,
+):
+    # SETUP
+    # Set environment variable SPLIT_MULTIPAGE_DOC
+    if split_multipage_doc_setup:
+        os.environ["SPLIT_MULTIPAGE_DOC"] = "1"
+    else:
+        del os.environ["SPLIT_MULTIPAGE_DOC"]
```

Comment:
You should never mock env vars like that (any mutable object). It brings a lot of side effects, you don't want to mess with that later.
Check out this
https://docs.python.org/3/library/unittest.mock.html#patch-dict